### PR TITLE
Add support for job labels

### PR DESCRIPTION
**Pull request by rustyconover**
Add support for job labels.

### CODE REVIEW

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java [hunk 1]
```diff
@@ -213,6 +215,44 @@ public Builder setEnv(Map<String,String> enviroment) {
...
+        /**
+         * Adds a collection of label to the job.
+         *
+         * This resets the labels of the job; it will remove labels that were previously set.
+         *
+         * @param labels specifies the labels to set for this job.
+         * @return this builder
+         */
+        public Builder setLabels(Map<String,String> labels) {
+            _labels = ImmutableMap.copyOf(labels);
```
Comment (leifwalsh) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: we won't be able to do setLabels() followed by addLabel(), I assume that's ok

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java [hunk 2]
```diff
@@ -371,6 +413,7 @@ private Job(UUID uuid, String name, String command, Double memory, Double cpus,
         } else {
             _container = null;
         }
+        _labels = ImmutableMap.copyOf(labels);
```
Comment (leifwalsh) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: similarly, we won't be able to initialize with a label map and then add additional labels

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java (hunk 1 above)
Comment (rustyconover) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: This is the same restrictions as the environment, so I believe we're okay.

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java (hunk 2 above)
Comment (rustyconover) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: Same reply as above.

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java [hunk 3]
```diff
@@ -507,6 +558,7 @@ public static JSONObject jsonizeJob(Job job)
         object.put("max_retries", job.getRetries());
         object.put("status", job.getStatus());
         object.put("env", env);
+        object.put("labels", env);
```
Comment (wyegelwel) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: Looks like we are setting the `"labels"` to `env`

File: jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java [hunk 4]
```diff
@@ -617,6 +669,16 @@ public static JSONObject jsonizeJob(Collection<Job> jobs)
                 }
                 jobBuilder.setEnv(envMap);
             }
+            if (json.has("labels")) {
+                JSONObject labelsJson = json.getJSONObject("labels");
+                Map<String, String> labelsMap = new HashMap<>();
+                if (labelsJson.length() > 0) {
```
Comment (wyegelwel) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: Is this check necessary?

Comment (rustyconover) [inline jobclient/src/main/java/com/twosigma/cook/jobclient/Job.java]: I think so, I'm just following how the environment is handled.

**Pull request merged**